File: src/ProjectR.Scripting/content/scripts/spells/RavagingFlow.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Ravaging Flow"
end

function GetDescription()
    return "Fires a concentrated laser of pure arcane engery at the target.\nUses up all of your MP, making the spell stronger.\nYou are completely exhausted after using this."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .7
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Pure
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  a:UseMP(70)
  local currentMP = a:GetCurrentMP()
  local damage = ((8 + currentMP/10) * aMD * (aARC/100) - 0.5 * dMR) * (100/dARC)  
  a:UseMP(300)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/RavagingFlow.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class RavagingFlow : SpellScriptBase
    {
        public override string Name { get { return "Ravaging Flow"; } }
        public override string Description { get { return "Fires a concentrated laser of pure arcane engery at the target.\nUses up all of your MP, making the spell stronger.\nYou are completely exhausted after using this."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] {EleMastery.ARC,  }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 70; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damageMod = caster.CurrentMP / 10;
            var damage = ((8 + damageMod) * aMD * (aARC/100) - .5 * dMR) * (100/dARC);
            caster.UseMP(200);
            DealDamage(damage);
        }
    }
}